pkg/agent/billing/billing.go

```diff
@@ -174,15 +175,14 @@ func runBillingMetricsCollector(
 		case <-collectTicker.C:
 			logger.Info("Collecting billing state")
 			if store.Stopped() && ctx.Err() == nil {
-				err := errors.New("VM store stopped but background context is still live")
-				logger.Panic("Validation check failed", zap.Error(err))
+				return errors.New("VM store stopped but background context is still live")
```

Comment:
IMO this should still be a panic. It's behavior that is never expected to occur that indicates a deeper issue somewhere else

Reply:
I am skeptical of ever using panic.

But what does this actually change, if the panic will be handled higher up the stack?

Reply:
My thinking is that "just" returning an error makes it seem like this issue is less severe, whereas a panic indicates that some invariant has been violated.

For this particular case, IIRC there is no expected outcome where the VM store stops while the autoscaler-agent is still running.

Thoughts?

Reply:
Ok, brought back the panic in 38c5512.